.github/workflows/paralleltest.yml

```diff
@@ -1,77 +1,174 @@
-
-      - name: Archive parallelList - extended.openjdk
-        uses: actions/upload-artifact@v2
-        with:
-          name: parallelList.mk
-          path: ${{ github.workspace }}/aqa-tests/TKG/parallelList.mk
-          
-  run-aqa-parallel:
-    runs-on: ${{ inputs.runner }}
-    if: ${{ inputs.runParallel == true }}
-    needs: setup-parallel
-    name: Run AQAvit ${{ inputs.runner }} - extended.openjdk
-    strategy:
-      fail-fast: false
-      matrix:
-        test_list: ['testList_0', 'testList_1', 'testList_2', 'testList_3', 'testList_4'] # numlist is hardcoded   
-    steps: 
-      - name: Download parallelList - extended.openjdk
-        uses: actions/download-artifact@v2
-        with:
-          name: parallelList.mk
-          path: ./
-
-      - name: Run AQA Parallel Test - extended.openjdk_${{ matrix.test_list }}
-        uses: adoptium/run-aqa@v2
-        with:
-          jdksource: 'customized'
-          customizedSdkUrl: ${{ inputs.customizedSdkUrl }}
-          aqa-testsRepo: 'adoptium/aqa-tests:v0.9.6-release' 
-          build_list: openjdk
-          target : -f parallelList.mk ${{ matrix.test_list }}
-
-      - uses: actions/upload-artifact@v2
-        if: always() # Always run this step (even if the tests failed)
-        with:
-          name: test_output_extended.openjdk_${{ matrix.test_list }}
-          path: ./**/output_*/*.tap
+name: Run AQAvit Paralleltest
+
+on:
+  workflow_dispatch: # Allows the job to be manually triggered
+    inputs:
+      customizedSdkUrl:
+        description: SDK URL
+        required: true
+      aqatest-repo:
+        description: adoptium/aqa-tests repository
+        required: true
+        type: string
+        default: 'adoptium/aqa-tests:v0.9.6-release'
+      testImage:
+        description: Native Libs URL
+        required: true
+      system:
+        description: System being tested
+        required: true
+      pool:
+        description: runner
+        required: true
+        type: string
+        default: None
+
+env:  # Links to the JDK build under test and the native test libs
+  USE_TESTENV_PROPERTIES: true
+
+jobs:
+  configuration:
+    runs-on: ubuntu-latest
+    outputs:
+      agent-config: ${{ steps.selection.outputs.config }}
+    steps:
+      - name: Select agent
+        id: selection
+        run: |
+          if [[ ${{ inputs.pool }} == "macos-x64" ]]; then
+            echo "config='["macos-10.15"]'" >> $GITHUB_OUTPUT
+          fi
+          if [[ ${{ inputs.pool }} == "macos-arm64" ]]; then
+            echo "config='["self-hosted", "macos", "ARM64"]'" >> $GITHUB_OUTPUT
+          fi
+          if [[ ${{ inputs.pool }} == "*windows2019-x64*" ]] || [[ ${{ inputs.pool }} == "*windows2022-aarch64*" ]] || [[ ${{ inputs.pool }} == "*ubuntu20.04-x64*" ]] || [[ ${{ inputs.pool }} == "*ubuntu20.04-aarch64*" ]]; then
+            echo "config='["self-hosted", "1ES.Pool=${{ inputs.pool }}"]'" >> $GITHUB_OUTPUT
+          fi

```

Comment:
I like this idea. The reasoning why I went and wrote a single job and step for this was to reduce the complexity of dependent jobs. If we split up into multiple jobs (which would greatly simplify this bit of code) the dependent jobs would most likely need to be broken up. There is one job to run the aqa tests and this would need to be split based on which agent to use. 

```yaml
jobs:
  configuration_win:
    if: ${{ inputs.system }} == "windows"...
  configuration_mac:
    if: ${{ inputs.system }} == "mac"...
  configuration_linux:
    if: ${{ inputs.system }} == "mac"...
....

  run-aqa-parallel-win:
    needs: configuration_win
    runs-on:  ${{ needs.configuration_win.outputs.agent}}
    .....
  run-aqa-parallel-mac:
    needs: configuration_mac
    runs-on: ${{ needs.configuration_mac.outputs.agent}}
    .....
  run-aqa-parallel-linux:
    needs: configuration_linux
    runs-on:  ${{ needs.configuration_linux.outputs.agent}}
    .....
```

I believe this to be the case since GitHub actions does not make it simple to choose which runner (`runs-on:`) in a somewhat dynamic fashion (for better or worse). At least for our case where we wish to use self-hosted agents and pools. This would be rather simple using GitHub agents since we could then supply which agent to use based on inputs.